packages/mui-material/src/Tooltip/Tooltip.js

```diff
@@ -669,72 +681,73 @@ const Tooltip = React.forwardRef(function Tooltip(inProps, ref) {
 
+    if (resolvedPopperProps?.popperOptions?.modifiers) {
+      tooltipModifiers = tooltipModifiers.concat(resolvedPopperProps.popperOptions.modifiers);
+    }
+
     return {
       ...PopperProps.popperOptions,
+      ...resolvedPopperProps?.popperOptions,
       modifiers: tooltipModifiers,
     };
-  }, [arrowRef, PopperProps]);
-
-  const ownerState = {
-    ...props,
-    isRtl,
-    arrow,
-    disableInteractive,
-    placement,
-    PopperComponentProp,
-    touch: ignoreNonTouchEvents.current,
-  };
+  }, [arrowRef, PopperProps.popperOptions, resolvedPopperProps?.popperOptions]);
 
   const classes = useUtilityClasses(ownerState);
-
-  const PopperComponent = slots.popper ?? components.Popper ?? TooltipPopper;
-  const TransitionComponent =
-    slots.transition ?? components.Transition ?? TransitionComponentProp ?? Grow;
-  const TooltipComponent = slots.tooltip ?? components.Tooltip ?? TooltipTooltip;
-  const ArrowComponent = slots.arrow ?? components.Arrow ?? TooltipArrow;
-
-  const popperProps = appendOwnerState(
-    PopperComponent,
-    {
-      ...PopperProps,
-      ...(slotProps.popper ?? componentsProps.popper),
-      className: clsx(
-        classes.popper,
-        PopperProps?.className,
-        (slotProps.popper ?? componentsProps.popper)?.className,
-      ),
+  const resolvedTransitionProps =
+    typeof slotProps.transition === 'function'
+      ? slotProps.transition(ownerState)
+      : slotProps.transition;
+  const externalForwardedProps = {
+    slots: {
+      popper: components.Popper,
+      transition: components.Transition ?? TransitionComponentProp,
+      tooltip: components.Tooltip,
+      arrow: components.Arrow,
+      ...slots,
+    },
+    slotProps: {
+      arrow: slotProps.arrow ?? componentsProps.arrow,
+      popper: { ...PopperProps, ...(resolvedPopperProps ?? componentsProps.popper) }, // resolvedPopperProps can be spread because it's already an object
+      tooltip: slotProps.tooltip ?? componentsProps.tooltip,
+      transition: {
+        ...TransitionProps,
+        ...(resolvedTransitionProps ?? componentsProps.transition),
+      },
     },
+  };
+
+  const [PopperSlot, popperSlotProps] = useSlot('popper', {
+    elementType: TooltipPopper,
+    externalForwardedProps,
     ownerState,
-  );
+    className: clsx(classes.popper, PopperProps?.className),
+  });
 
-  const transitionProps = appendOwnerState(
-    TransitionComponent,
-    { ...TransitionProps, ...(slotProps.transition ?? componentsProps.transition) },
+  const [TooltipTransitionSlot, transitionSlotProps] = useSlot('transition', {
+    elementType: Grow,
+    externalForwardedProps,
     ownerState,
-  );
+  });
 
-  const tooltipProps = appendOwnerState(
-    TooltipComponent,
-    {
-      ...(slotProps.tooltip ?? componentsProps.tooltip),
-      className: clsx(classes.tooltip, (slotProps.tooltip ?? componentsProps.tooltip)?.className),
-    },
+  const [TooltipTooltipSlot, tooltipSlotProps] = useSlot('tooltip', {
```

Comment:
The naming convention seems to be?

```suggestion
  const [TooltipSlot, tooltipSlotProps] = useSlot('tooltip', {
```

x the other components